main_test.go

```diff
@@ -428,9 +484,10 @@ func TestLifecycle(t *testing.T) {
 						} else {
 							body = respBody[i : i+5]
 						}
+						expectRejct := (i+5 >= responseBodyLimit) && responseBodyLimitSlice != nil && strings.Contains(tt.inlineRules, "SecResponseBodyLimitAction Reject")
```

Comment:
I would avoid doing all this conditional in the tests and instead adding a new field expectReject otherwise the test itself becomes a source of bugs.

Reply:
Done, it makes the test a bit less flexible, but I agree that all that complexity for an individual test case is not worth it